WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/stock/ProductStockRepository.kt

```diff
@@ -34,22 +37,42 @@ class ProductStockRepository @Inject constructor(
             if (stockReportResult.isError) {
                 Result.failure(WooException(stockReportResult.error!!))
             } else {
+                if (isCachedStockTheSame(stockReportResult.model, stockStatus)) {
+                    return Result.success(cachedStockReport[stockStatus]!!)
+                }
+
                 val (productSalesResult, variationSalesResult) = getProductSalesReports(stockReportResult.model!!)
                 when {
                     productSalesResult.isError -> Result.failure(WooException(productSalesResult.error))
                     variationSalesResult.isError -> Result.failure(WooException(variationSalesResult.error))
-                    else ->
-                        Result.success(
-                            mapToProductStockItems(
-                                stockReportResult.model!!,
-                                productSalesResult.model!! + variationSalesResult.model!!
-                            )
+                    else -> {
+                        val report = mapToProductStockItems(
+                            stockReportResult.model!!,
+                            productSalesResult.model!! + variationSalesResult.model!!
                         )
+                        cachedStockReport[stockStatus] = report
+                        Result.success(report)
+                    }
                 }
             }
         }
     }
 
+    private fun isCachedStockTheSame(
+        fetchedStock: Array<ProductStockItemApiResponse>?,
+        stockStatus: ProductStockStatus
+    ): Boolean =
+        fetchedStock?.let { fetchedStockItems ->
+            val cachedStock = cachedStockReport[stockStatus] ?: return false
+            if (cachedStock.isEmpty()) return false
```

Comment:
When testing this, I had a case where I added a new product with low stock, but it didn't show up even after pull to refreshes, and the case is that we check just against the existing items in the cache below, so this function returns `true` when we add new items.

To solve this, I suggest applying the following shortcut to return cancel when the sizes are different:
```suggestion
            if (cachedStock.isEmpty() || fetchedStockItems.size != cachedStock.size) return false
```